flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/data/RowDataUtil.java

```diff
@@ -79,7 +79,11 @@ public static Object convertConstant(Type type, Object value) {
    * the arity check.
    */
   public static RowData clone(
-      RowData from, RowData reuse, RowType rowType, TypeSerializer[] fieldSerializers) {
+      RowData from,
```

Comment:
Please create a deprecated version of the original public method to keep compatibility.

Reply:
Thanks for the reminder! I've added the original function below. The reason I didn't directly use the new function is that it requires FieldGetter for all fields, while the old function only needs to construct FieldGetter for fields with values. Constructing FieldGetter for all fields might negatively impact performance.

Reply:
I feel that performance degradation is acceptable for the deprecated methods if it is justified otherwise. The number of the duplicated lines seems high enough for me to try to avoid them.

The users could always do the caching of the getters themselves, or find another solution soon, since we remove the deprecated methods regularly.

WDYT?

Also CC-ed, @stevenzwu, who might be also interested in this PR

Reply:
OK